functions/self_update.go

```diff
@@ -0,0 +1,43 @@
+package functions
+
+import (
+	"log"
+	"strings"
+
+	"github.com/blang/semver"
```

Comment:
https://github.com/hashicorp/go-version is more popular and more recently updated go version library.  Already used in netmaker branch netclient_refactor

Reply:
But the `go-github-selfupdate` library only works with `github.com/blang/semver`. Would need to fork the repo and refactor it to use `hashicorp/go-version`

Reply:
ok, didn't realize that